### PR TITLE
Make postconf/newaliases respect file type

### PR DESCRIPTION
**Pull request by EvaSDK**
As reported in issue #50, formula needs to respect the file type in order to run postmap/newaliases only when appropriate.

Added support for `default_database_type` while at it. The default value is just the one I found to be the most common but this could be customized via `map.jinja` or determined automatically by calling `postconf -d` however I decided to discard that until someone requests it eventually.

Things that could be done to enhance alias database handling is override `map.jinja` `aliases_file` with pillar provided `alias_database` setting and support running `postalias` on `alias_maps` if it makes any difference wrt `postmap`.

### CODE REVIEW

**Comment by aboe76** [comment]
@EvaSDK can this be merged?

**Comment by duk3luk3** [comment]
It needs review/testing.

**Comment by EvaSDK** [comment]
ping ?

File: postfix/init.sls [hunk 1]
```diff
@@ -12,9 +12,22 @@ postfix:
...
+  {%- if ':' in file_path %}
+    {%- set file_type, file_path = postfix.aliases_file.split(':') %}
+  {%- else %}
+    {%- set file_type = default_database_type %}
+  {%- endif %}
+  {%- if file_type in ("btree", "cdb", "dbm", "hash", "sdbm") %}
+    {%- set need_newaliases = True %}
+  {%- endif %}
+postfix_alias_database:
   file.managed:
```
Comment (aboe76) [inline postfix/init.sls]: missing 
```
- name: {{ file_path }}

Comment (EvaSDK) [inline postfix/init.sls]: thanks for catching this.

**Comment by aboe76** [comment]
@duk3luk3 are you happy? or do you want to test this?

**Comment by duk3luk3** [comment]
Looks good to me.

@EvaSDK could you please squash those last two commits into the preceding one?

**Comment by EvaSDK** [comment]
done.

**Pull request merged**